level2/restaurants/gabi_restaurant.rb

```diff
@@ -0,0 +1,86 @@
+
+require 'pry'
+
+UNAVAILABLE_ITEM_ERROR = 'Items not available'
+DOUBLE_DISCOUNT_ERROR = 'Double discount is not allowed'
+
+class Restaurant
+
+    def initialize(name,*items)
+        @name = name.to_s
+        @items = items
+    end
+        
+    def name
+        @name
+    end
+
+    def items
+        @items
+    end
+
+    def add_items(*item)
+        item.each{|i| @items << i}
+    end
+
+    def remove_item(item_name)
+        item = @items.find{|item| item.name == item_name}
+        @items.delete(item)
+    end
+
+    def place_order(ordered_items, discount: 0)
+        order_total = 0
+        ordered_items.each do |ordered_item| 
+            #binding.pry
+            item = @items.find{|item| item.name == ordered_item[:name]}
+            return UNAVAILABLE_ITEM_ERROR if item.nil?
+
+            price = item.price
+            stock = item.stock
+            quantity = ordered_item[:count]
+            if stock >= quantity
+                item_discount = ordered_item[:discount] || 0
+                return DOUBLE_DISCOUNT_ERROR if discount != 0 && item_discount != 0
+
+                total = price * quantity * (1 - item_discount)
+                order_total += total
+                
+                item.deplete_stock(quantity)
```

Comment:
this is not in the tests, but what happens when we order two items and the first one is in stock while the second one is not?